apps/prairielearn/src/models/course-request.ts

```diff
@@ -0,0 +1,99 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { InstitutionSchema, DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z.object({
+  authn_user_id: IdSchema.optional(),
+  finish_date: DateFromISOString.optional(),
+  id: IdSchema.optional(),
+  number: z.number().optional(),
+  start_date: DateFromISOString.optional(),
+  status: z.string().optional(),
+});
+
+const CourseRequestRowSchema = z.object({
+  approved_by_name: z.string().nullable(),
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  jobs: JobsRowSchema,
+  last_name: z.string().nullable(),
+  short_name: z.string().nullable(),
+  status: z.enum(['pending', 'approved', 'denied', 'creating', 'failed']),
+  title: z.string().nullable(),
+  user_id: IdSchema.nullable(),
+  user_name: z.string().nullable(),
+  work_email: z.string().nullable(),
+});
+
+export async function getCourseRequests(show_all: boolean) {
+  return await queryRow(
+    sql.get_requests,
+    { show_all },
+    z.object({
+      institutions: z.array(InstitutionSchema),
+      course_requests: z.array(CourseRequestRowSchema),
+    }),
+  );
+}
+
+export async function updateCourseRequest(req, res) {
```

Comment:
As a rule, we don't want functions that take `req`/`res` arguments to be present in `models/`; models should typically deal strictly with pre-validated data. Consider moving these to a `lib/course-requests.ts` file

Reply:
These have been moved to `lib/course-requests.ts` in commit [46a7046](https://github.com/PrairieLearn/PrairieLearn/pull/9238/commits/46a7046a6311dae7797f62cb4227fcbf1a3ead90).